### PR TITLE
feat: Add new Pull Request template for SAP Cloud SDK Java repository

### PR DESCRIPTION
**Pull request by CharlesDuboisSAP**
# Pull Request Description 🤖

This pull request addresses backlog item [#316](https://github.com/SAP/cloud-sdk-java-backlog/issues/316). The goal of this PR is to introduce a new Pull Request template for the SAP Cloud SDK Java repository.

## Changes

The changes in this PR are as follows:

- A new file `.github/PULL_REQUEST_TEMPLATE.md` has been added. This file contains the template that will be used for all future pull requests in this repository. The template includes sections for providing context, describing the feature scope, stating the release timeline, and defining the completion criteria for the PR.

## Context

This change is part of an effort to standardize and improve the quality of pull requests in the SAP Cloud SDK Java repository. By providing a template, we aim to make it easier for contributors to provide all necessary information when creating a PR, and for reviewers to understand the changes being proposed.

## Link to Backlog Item

You can find the backlog item associated with these changes [here](https://github.com/SAP/cloud-sdk-java-backlog/issues/316).

## Instructions for Reviewers

Please review the new PR template and provide feedback on its structure and content. Consider whether it includes all necessary sections and whether the instructions are clear and easy to follow.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+By default some items are marked not relevant because we don't need them frequently. Please still consider if they apply for your PR.
+-->
+
+- [ ] Functionality scope stated & covered
+- [ ] Tests cover the scope above
+- [ ] ~Error handling created / updated & covered by the tests above~
+- [ ] ~Documentation updated~
+- [ ] ~Release notes updated~
+- [ ] ~Maintained traceability mapping~
```
Comment (Johannes-Schneider) [inline .github/PULL_REQUEST_TEMPLATE.md:47]: We are no longer doing traceability mapping, right?
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+<!--
+An example DoD that is not yet completed might look like this:
+
+- [x] Functionality scope stated & covered
+- [ ] Tests created / updated _according to the scope_ above
+- [x] ~Error handling created / updated & covered by the tests above~
+- [x] ~Documentation updated~
+- [ ] ~Release notes updated~
+- [x] ~Maintained traceability mapping~
```
Comment (Johannes-Schneider) [inline .github/PULL_REQUEST_TEMPLATE.md:57]:
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+Which would mean:
+
+> I implemented the functionality and declared the scope of my changes in the description above. 
+> I still have to add some tests but don't have to change any error handling or documentation.
+> However, I have not yet considered if we need release notes. 
+> But we don't need to have a traceability mapping.
```
Comment (Johannes-Schneider) [inline .github/PULL_REQUEST_TEMPLATE.md:64]:
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (CharlesDuboisSAP) [reply .github/PULL_REQUEST_TEMPLATE.md:47]: Yes we do, `module-inventory.json` is used by our GitHub actions.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+<!-- 
+Do you want this change to be part of a certain release?
+If so, please state the release hereafter (replace `XY` with the release number).
+Otherwise, this change gets merged and thereby released as time permits.
+-->
+
+_This PR is relevant for the release version XY._
+
+
```
Comment (newtork) [inline .github/PULL_REQUEST_TEMPLATE.md:33]: Do we still need that?
I didn't find it useful 95% of times.

Comment (CharlesDuboisSAP) [reply .github/PULL_REQUEST_TEMPLATE.md]: Removed

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (newtork) [reply .github/PULL_REQUEST_TEMPLATE.md]: "traceability" is the process of keeping track of `BLI/issue <-> unit test`. See `traceability/requirement.mapping.json` in our legacy repo.

I don't think it translates to `module-inventory.json` 🤔
I wouldn't immediately make the file part of **all** PRs to consider. I think.

Comment (CharlesDuboisSAP) [reply .github/PULL_REQUEST_TEMPLATE.md]: Removed

**Review by Johannes-Schneider — APPROVED** [review]

**Pull request merged**